cmd/vic-machine/validator.go

```diff
@@ -0,0 +1,210 @@
+	v.ClusterPath = strings.Split(input.computeResourcePath, "/Resources")[0]
+
+	if v.ClusterPath == "" {
+		err := errors.Errorf("Could not determine cluster from specified -compute path, %s", input.computeResourcePath)
+		return nil, err
+	}
+
+	v.ResourcePoolPath = input.computeResourcePath
+	v.ImageStorePath = fmt.Sprintf("/%s/datastore/%s", v.DatacenterName, input.imageDatastoreName)
+
+	v.TargetPath = fmt.Sprintf("%s:%s@%s/sdk", input.user, *input.passwd, input.target)
+	vchConfig.Target = v.TargetPath
+	vchConfig.Insecure = input.insecure
+
+	v.ExternalNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.externalNetworkName)
+	v.BridgeNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.bridgeNetworkName)
+	v.BridgeNetworkName = input.bridgeNetworkName
+	if input.managementNetworkName != "" {
+		v.ManagementNetworkPath = fmt.Sprintf("/%s/network/%s", v.DatacenterName, input.managementNetworkName)
+		v.ManagementNetworkName = input.managementNetworkName
+	}
+	vchConfig.ImageStoreName = input.imageDatastoreName
+	vchConfig.DatacenterName = v.DatacenterName
+	vchConfig.ClusterPath = v.ClusterPath
+
+	if err := v.validateConfiguration(input, vchConfig); err != nil {
+		return nil, err
+	}
+	return vchConfig, nil
+}
+
+func (v *Validator) validateConfiguration(input *Data, vchConfig *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infof("Validating supplied configuration")
+
+	var err error
+	v.Context = context.TODO()
+	sessionconfig := &session.Config{
+		Service:        v.TargetPath,
+		Insecure:       input.insecure,
+		DatacenterPath: v.DatacenterName,
+		ClusterPath:    v.ClusterPath,
+		DatastorePath:  v.ImageStorePath,
+		PoolPath:       v.ResourcePoolPath,
+	}
+
+	v.Session, err = session.NewSession(sessionconfig).Create(v.Context)
+	if err != nil {
+		log.Errorf("Failed to create session: %s", err)
+		return err
+	}
+	_, err = v.Session.Connect(v.Context)
+	if err != nil {
+		return err
+	}
+
+	if _, err = v.Session.Populate(v.Context); err != nil {
+		log.Errorf("Failed to get resources: %s", err)
+		return err
+	}
+
+	// find the host(s) attached to given storage
+	if _, err = v.Session.Datastore.AttachedClusterHosts(v.Context, v.Session.Cluster); err != nil {
+		log.Errorf("Unable to get the list of hosts attached to given storage: %s", err)
+		return err
+	}
+
+	if err = v.createBridgeNetwork(); err != nil && !input.force {
+		return errors.Errorf("Creating bridge network failed with %s", err)
+	}
+
+	if err = v.setNetworks(vchConfig); err != nil {
+		return errors.Errorf("Find networks failed with %s", err)
+	}
+	vchConfig.ComputeResources = append(vchConfig.ComputeResources, v.Session.Pool)
+	vchConfig.ImageStores = append(vchConfig.ImageStores, v.ImageStorePath)
+	//TODO: Add more configuration validation
+	return nil
+}
+
+func (v *Validator) getNetworkPath(net object.NetworkReference) (string, string, error) {
+	switch t := net.(type) {
+	case *object.DistributedVirtualPortgroup:
```

Comment:
Use `t` instead of `net` inside the case statements. `t` is already of the appropriate type as per the type switch semantics:
https://golang.org/doc/effective_go.html#type_switch

```
switch t := net.(type) {
   case *object.DistributedVirtualPortgroup:		
      return t.InventoryPath, t.Name(), nil
   case *object.Network:
      return t.InventoryPath, t.Name(), nil
   default:
      return "", "", errors.Errorf("Unknown network card type: %s", reflect.TypeOf(t))
}
```

You can also add a test for the raw dvswitch types - I think this is likely to be a common enough mistake that it warrants a custom error message:
DistributedVirtualSwitch
VmwareDistributedVirtualSwitch